src/utils/send.js

```diff
@@ -1,35 +1,61 @@
 "use strict";
 
-const https = require('https');
-
-// `data` is a JSON-stringified object.
-function httpsSend(data, requestConfigs) {
-  const req = https.request(requestConfigs, (response) => {
-    let responseData = '';
-    response.on('data', (chunk) => responseData += chunk);
-    response.on('end', () => console.log('Response body: ', responseData));
-  });
-
-  req.on('error', (err) => console.error('Error: ', err));
-  req.write(data);
-  req.end();
+import axios from "axios";
+import processEvent from "./process-event.js";
+import processUser from "./process-user.js";
+import getTimeStamp from "./get-timestamp.js";
+import stringifyObject from "./process-attributes.js";
+
+async function put(category, data, requestConfigs) {
+  try {
+    let formatted;
+    if (category === "event") {
+      formatted = processEvent(data);
+    } else if (category === "user") {
+      formatted = processUser(data);
+    } else {
+      throw new Error("Invalid category specified");
+    }
+
+    const response = await axios.post(
+      `https://${requestConfigs.host}${requestConfigs.path}`,
+      formatted,
+    );
+
+    return response.data;
+  } catch (error) {
+    console.error(error);
+    return error;
+  }
+}
+
+async function update(data, requestConfigs) {
```

Comment:
Has this action been tested and proven to correctly update user data?

Reply:
Yes